persistent/Database/Persist/Sql/Orphan/PersistQuery.hs

```diff
@@ -225,18 +228,19 @@ dummyFromFilts _ = Nothing
 
 getFiltsValues :: forall val. (PersistEntity val, PersistEntityBackend val ~ SqlBackend)
                => SqlBackend -> [Filter val] -> [PersistValue]
-getFiltsValues conn = snd . filterClauseHelper False False conn OrNullNo
+getFiltsValues conn = snd . filterClauseHelper False False False conn OrNullNo
 
 data OrNull = OrNullYes | OrNullNo
 
 filterClauseHelper :: (PersistEntity val, PersistEntityBackend val ~ SqlBackend)
              => Bool -- ^ include table name?
              -> Bool -- ^ include WHERE?
+             -> Bool -- ^ use postgresl EXCLUDE
              -> SqlBackend
              -> OrNull
              -> [Filter val]
              -> (Text, [PersistValue])
-filterClauseHelper includeTable includeWhere conn orNull filters =
+filterClauseHelper includeTable includeWhere includeExcluded conn orNull filters =
```

Comment:
I added this kinda ratchet param because the filterClause was breaking on the `EXCLUDED` value for PostgreSQL again.  See the next comment for where I use it...

Reply:
Yeah, I agree this is a bit dicey.

`Bool`ean blindness is a problem here, and since this is the first time we're 'exposing' this function, we can fix that.

Instead of `includeExcluded :: Bool` and `includeTable :: Bool` (note that `includeTable = True` means that `includeExcluded` *doesn't matter*!), let's make a new type:

```haskell
data FilterTablePrefix
    = PrefixTableName -- ^ Include the table name
    | PrefixExcluded -- ^ Valid only on Postgresql
```

Then we pass a `Maybe FilterTablePrefix` instead of the two `Bool` arguments - `Nothing` corresponds to `False, False`.

Reply:
this is a great way to data model